chains/evm/contracts/test/libraries/Internal.t.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {Internal} from "../../libraries/Internal.sol";
+import {InternalTestHelper} from "../helpers/InternalTestHelper.sol";
+import {Test} from "forge-std/Test.sol";
+
+contract InternalSetup is Test {
+  address internal constant VALID_EVM_ADDRESS = 0x1234567890123456789012345678901234567890;
+  address internal constant PRECOMPILE_ADDRESS = address(0x01);
+  bytes internal constant VALID_32_BYTE_ADDRESS =
+    abi.encode(uint256(0x1234567890123456789012345678901234567890123456789012345678901234));
+  bytes internal constant VALID_TVM_ADDRESS =
+    hex"11ff1234567890123456789012345678901234567890123456789012345678901234abcd";
+  bytes internal constant INVALID_TVM_ADDRESS_ZERO_ACCOUNT =
+    hex"11ff000000000000000000000000000000000000000000000000000000000000000012ab";
+
+  InternalTestHelper internal s_helper;
+
+  function setUp() public virtual {
+    s_helper = new InternalTestHelper();
+  }
+}
+
+contract Internal_validateEVMAddress is InternalSetup {
+  function test_validateEVMAddress_succeeds_onValidAddress() public {
+    bytes memory validAddress = abi.encode(VALID_EVM_ADDRESS);
+    s_helper.validateEVMAddress(validAddress);
+  }
+
+  function test_validateEVMAddress_reverts_onInvalidLength() public {
+    bytes memory invalidAddress = new bytes(31);
+    vm.expectRevert(abi.encodeWithSelector(Internal.InvalidEVMAddress.selector, invalidAddress));
+    s_helper.validateEVMAddress(invalidAddress);
+  }
+
+  function test_validateEVMAddress_reverts_onPrecompileAddress() public {
+    bytes memory precompileAddress = abi.encode(PRECOMPILE_ADDRESS);
+    vm.expectRevert(abi.encodeWithSelector(Internal.InvalidEVMAddress.selector, precompileAddress));
+    s_helper.validateEVMAddress(precompileAddress);
+  }
+
+  function test_validateEVMAddress_reverts_onOversizedAddress() public {
+    bytes memory invalidAddress = abi.encode(uint256(type(uint160).max) + 1);
+    vm.expectRevert(abi.encodeWithSelector(Internal.InvalidEVMAddress.selector, invalidAddress));
+    s_helper.validateEVMAddress(invalidAddress);
+  }
+
+  function test_validateEVMAddress_succeeds_onBoundaryAddresses() public {
+    bytes memory lowerBoundary = abi.encode(uint256(Internal.EVM_PRECOMPILE_SPACE));
+    s_helper.validateEVMAddress(lowerBoundary);
+
+    bytes memory upperBoundary = abi.encode(uint256(type(uint160).max));
+    s_helper.validateEVMAddress(upperBoundary);
+  }
+}
+
+contract Internal_validate32ByteAddress is InternalSetup {
```

Comment:
We don't do multiple contracts per file, please split each contract into its own file